californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -175,6 +181,20 @@ public UdpMatcher(final NetworkConfig config) {
 
 		exchangesByMID.put(idByMID, exchange);
 		exchangesByToken.put(idByToken, exchange);
+
+		if(request.isMulticast()) {
+			if(executor != null) {
```

Comment:
no need to check for null. We have already checked for that in `UdpMatcher.start()`